File: src/test/java/com/github/xumeng/java/problems/easy/problem0278/Solution0278Test.java
package com.github.xumeng.java.problems.easy.problem0278;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class Solution0278Test {

    private final Solution0278 solution = new Solution0278();

    @Test
    public void test1() {
        int n = 5;
        int expected = 4;
        solution.setBadVersion(expected);
        int bad = solution.firstBadVersion(n);
        assertEquals(expected, bad);
    }

    @Test
    public void test2() {
        int n = 1;
        int expected = 1;
        solution.setBadVersion(expected);
        int bad = solution.firstBadVersion(n);
        assertEquals(expected, bad);
    }

    @Test
    public void test3() {
        int n = 2147483647;
        int expected = 100;
        solution.setBadVersion(expected);
        int bad = solution.firstBadVersion(n);
        assertEquals(expected, bad);
    }

    @Test
    public void test4() {
        int n = 2147483647;
        int expected = 2147483547;
        solution.setBadVersion(expected);
        int bad = solution.firstBadVersion(n);
        assertEquals(expected, bad);
    }

}

File: src/test/kotlin/com/github/xumeng/kotlin/problems/easy/problem0278/Solution0278Test.kt
package com.github.xumeng.kotlin.problems.easy.problem0278

import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.Test

class Solution0278Test {

    private val solution = Solution0278()

    @Test
    fun test1() {
        val n = 5
        val expected = 4
        solution.setBacVersion(expected)
        val bad = solution.firstBadVersion(n)
        assertEquals(expected, bad)
    }

    @Test
    fun test2() {
        val n = 1
        val expected = 1
        solution.setBacVersion(expected)
        val bad = solution.firstBadVersion(n)
        assertEquals(expected, bad)
    }

    @Test
    fun test3() {
        val n = 2147483647
        val expected = 100
        solution.setBacVersion(expected)
        val bad = solution.firstBadVersion(n)
        assertEquals(expected, bad)
    }

    @Test
    fun test4() {
        val n = 2147483647
        val expected = 2147483547
        solution.setBacVersion(expected)
        val bad = solution.firstBadVersion(n)
        assertEquals(expected, bad)
    }

}